samples/cellular/modem_shell/src/ping/icmp_ping_shell.c

```diff
@@ -80,8 +80,9 @@ int icmp_ping_shell_th(const struct shell *shell, size_t argc, char **argv,
 	}
 #endif
 
-	/* Start from the 1st argument */
+	optreset = 1;
 	optind = 1;
+	int opt;
```

Comment:
While not directly related to this PR, I'm wondering if the parameters for the `ping` command should be changed. Currently, ping takes one "mandatory option" and "optional options":
```
Usage: ping [optional options] -d destination

mandatory options:
  -d, --destination, [str] Name or IP address
optional options:
  -t, --timeout, [int]     Ping timeout in milliseconds
...
```
I've been always wondering why the destination address is given using option `-d`. It's not an option, because it's not optional. The command could work in the same way as in Linux and Windows:
```
Usage
  ping [options] <destination>
```
```
Usage: ping [-t] [-a] [-n count] [-l size] [-f] [-i TTL] [-v TOS]
            [-r count] [-s count] [[-j host-list] | [-k host-list]]
            [-w timeout] [-R] [-S srcaddr] [-c compartment] [-p]
            [-4] [-6] target_name
```

Reply:
I don't know or remember but changing this makes the parsing a bit more complex. It's not something getopt_long returns then but you'll take the argument after the options.

Generally, I'm neutral on this comment as I'm not changing it in the context of this PR.

@jhirsi, thoughts?

Reply:
Plus, it's a change in test scripts.

Reply:
This change would be something like this after getopt loop (plus error handling from `-d`):

```
	if (optind < argc) {
		printf("argv[%d]=%s\n", optind, argv[optind]);
		strcpy(ping_args.target_name, argv[optind]);
	}

```

Reply:
Yes, it would make parsing the command line more complicated. And would require changes in tests. If we don't want to go that way, I suggest we at least update the help text. For example:
```
Usage: ping [options] -d destination

  -d, --destination, [str] Name or IP address
options:
  -t, --timeout, [int]     Ping timeout in milliseconds
...
```

Reply:
Help changed.